.github/workflows/main.yml

```diff
@@ -16,6 +16,10 @@ jobs:
 
     steps:
     - uses: actions/checkout@v3
+    - name: SetupRust
+      uses: ATiltedTree/setup-rust@v1
```

Comment:
This action still uses Node.js 12, which has been [deprecated by GHA](https://github.blog/changelog/2022-09-22-github-actions-all-actions-will-begin-running-on-node16-instead-of-node12/) and can stop working at any moment.

I'd recommend [dtolnay/rust-toolchain](https://github.com/dtolnay/rust-toolchain) as an alternative.